src/content/docs/tanstack/tanstack-query.md

```diff
@@ -0,0 +1,250 @@
+Features such as automatic refetching, caching, query invalidation, and prefetching has made it a popular choice for developers looking to simplify their data fetching and state management needs.
+
+## History
+
+[![Star History Chart](https://api.star-history.com/svg?repos=tanstack/query&type=Date)](https://star-history.com/#tanstack/query&Date)
+
+### Origins
+
+The idea for TanStack Query came from wanting to move away from Redux in favour of a better, simpler way to manage asynchronous state in React applications. Tanner Linsley, the creator of the library, wanted to simplify the process of fetching, caching, and updating data in his own application, [Nozzle.io](https://nozzle.io/). Inspired by the introduction of the React hooks API, a replacement of the class based approach at the time, and the straightforwardness of `useState` and `useEffect` hooks, he aspired to create a similar experience for data fetching. The API was inspired by and built on the same design principles as [Draqula](https://github.com/vadimdemedes/draqula) by Vadim Demedes, a lightweight alternative to Apollo Client.
+
+### Philosophy
+
+TanStack Query was designed with these core principles in mind:
+
+- **Ergonomic**: The library should be easy to use and integrate into existing React applications.
+- **Simple**: The API should be straightforward and intuitive, with minimal boilerplate required.
+- **Forgiving**: The library should handle common use cases and edge cases gracefully, without requiring complex workarounds.
+- **Customizable**: Users should be able to customize and extend the library to suit a wide range of data fetching scenarios and application requirements.
+
+### Milestones
+
+- **V1 Release (26/02/2020):** Marked the debut of React Query, introducing the useQuery hook, providing a seamless way to fetch, cache, and update data.
+- **V2 Enhancements (22/06/2020):** Introduced boolean status indicators (isLoading, isSuccess, isError), the useMutation hook for data mutations, and useQueryClient for query client access.
+- **V3 Advancements (08/10/2020):** Brought SSR support, flexible caching, bi-directional infinite queries, and the useQueries hook for parallel queries.
+- **Rebranding to TanStack Query (17/07/2022):** With V4, React Query became TanStack Query, extending its core to support multiple frameworks beyond React, enhancing TypeScript support, and introducing a new plugin system.
+- **V5 Innovations (17/10/2023):** Focused on streamlining the API, improving TypeScript integration, and introducing features like optimistic updates and first-class support for React's suspense, emphasizing a smaller, more intuitive library.
+
+---
+
+## Why do we need TanStack Query?
+
+### The problem
```

Comment:
Great initiative for showing the problem and then presenting a solution. it is a great way of catching the interest of the reader.